resources/kcp/charts/component-reconcilers/templates/deployment.yaml

```diff
@@ -1,56 +1,59 @@
 {{ $global := . }}
-{{- range $component := .Values.global.components }}
-{{ $values := index $global.Values $component }}
-{{ if not (hasKey $global.Subcharts $component) }}
+{{- range $component := .Values.global.images.components }}
+{{ $values := index $global.Values $component.name }}
+{{ if not (hasKey $global.Subcharts $component.name) }}
 apiVersion: apps/v1
 kind: Deployment
 metadata:
   labels:
     kyma-project.io/component-reconciler: ""
-    component: {{ $component }}
-  name: {{ $component }}-reconciler
+    component: {{ $component.name }}
+  name: {{ $component.name }}-reconciler
   namespace: {{ $global.Release.Namespace }}
 spec:
   selector:
     matchLabels:
       kyma-project.io/component-reconciler: ""
-      component: {{ $component }}
+      component: {{ $component.name }}
   replicas: {{ default $global.Values.deployment.replicasCount (($values).deployment).replicasCount }}
   template:
     metadata:
       labels:
         kyma-project.io/component-reconciler: ""
-        component: {{ $component }}
+        component: {{ $component.name }}
     spec:
-      serviceAccountName: component-reconcilers-{{ $component }}
+      serviceAccountName: component-reconcilers-{{ $component.name }}
       nodeSelector:
         {{- toYaml $global.Values.deployment.nodeSelector | nindent 8 }}
       containers:
-      - image: "{{ $global.Values.global.images.component_reconciler }}"
+      - image: "{{ $.Values.global.images.reconciler_repo }}/component:{{ $component.version }}"
         imagePullPolicy: {{ $global.Values.deployment.imagePullPolicy }}
         args:
           - reconciler
           - start
-          - {{ $component }}
+          - {{ $component.name }}
           - --workspace=/tmp/reconciler
           {{- if $global.Values.options.verbose }}
           - --verbose
           {{- end }}
           - --worker-count={{ default $global.Values.options.workerCount (($values).options).workerCount }}
+          - --worker-timeout={{ default $global.Values.options.workerTimeout (($values).options).workerTimeout }}
```

Comment:
Overriding these for a component requires adding the following to the .Values at the very top-level of `values.yaml` file. For example, for Istio:
```
istio:
  options:
    workerCount: 1
    workerTimeout: 2
    retriesMax: 3
    retriesDelay: 4

global:
  ...
```
Is that on purpose? I'd say, a better place for that would be "closer" to the component entry, in the place where we define it's image tag - like this:

```
   components:
      - name: "base"
        version: "38b04f8c"
      - name: "busola-migrator"
        version: "38b04f8c"
      - name: "cleaner"
        version: "38b04f8c"
      - name: "connectivity-proxy"
        version: "38b04f8c"
      - name: "istio-configuration"
        version: "38b04f8c"
      - name: "istio"
        version: "38b04f8c"
        options:
          workerCount: 1
          workerTimeout: 2
          retriesMax: 3
          retriesDelay: 4
```

What do you think?

Reply:
List here some facts based on our discussion for later refactoring:
- basically the overwrite content which Tomasz mentioned has already been offered by [management-plane-config](https://github.tools.sap/kyma/management-plane-config/blob/master/resources/control-plane/config/stage/kcp.yaml), 
- The `components` entry is under `images` at the moment, my purpose here is easy for update image tag to keep in same block, if later needs to take this entry as an overall place to configure components, should be move outside
- Found a bug for rma overwrite value, a followup fix will be created.